core/src/modules/ops/record.ts

```diff
@@ -643,8 +643,11 @@ export namespace RecordOps {
 
     for (const _export of existingExportNotDeleted) {
       if (!destinations.map((d) => d.id).includes(_export.destinationId)) {
-        const destination = await Destination.findById(_export.destinationId);
-        if (destination.state === "ready") destinations.push(destination);
+        const destination = await Destination.findOne({
+          where: { id: _export.destinationId, state: "ready" },
+        });
+        // the destination may have been deleted or become not-ready
+        if (destination) destinations.push(destination);
```

Comment:
Hm.. why was this failing? The `findById` method on Destinations should clear the default scope and find the deleted one anyway (and then checked that it's ready by the conditional) 🤔

Reply:
I guess I don't quite understand how this specific code change addresses that since it seems functionally equivalent to me- it's still looking for that destination id and pushing it if it's ready

Reply:
Oh! `.findById` throws if the item can't be found - this where clause won't throw, and might find `undefined`

Reply:
There's also probably an optimization to INNER JOIN with destinations in the query above to skip this second lookup too

Reply:
oh derp, for some reason I thought the destination would be there but in the `deleted` state. Got it!